Sources/NIOCore/ByteBuffer-aux.swift

```diff
@@ -88,6 +88,24 @@ extension ByteBuffer {
         )
     }
 
+    // MARK: Hex encoded string APIs
+    /// Write `string` into this `ByteBuffer` in ASCII hexadecimal, moving the writer index forward appropriately.
+    /// - parameters:
+    ///     - plainHexEncodedBytes: The hex encoded string to write.
+    /// - returns: The number of bytes written.
+    @discardableResult
+    @inlinable
+    public mutating func writePlainHexEncodedBytes(_ plainHexEncodedBytes: String) throws -> Int {
+        var slice = plainHexEncodedBytes.utf8[...]
+        var written = 0
+
+        while let nextByte = try slice.popNextHexByte() {
```

Comment:
Will a throw here shut down the channel and mean the inconsistent state of the byte buffer with only half the values written is OK? 

OR Should I do an initial pass and validate BEFORE then going on and writing them to the buffer to ensure a consistent state? 

I did it initially the latter way, but poking around I assumed the channel closes and that bytebuffer is no longer used.